core-bundle/src/Resources/contao/widgets/RootPageDependentSelect.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao;
+
+class RootPageDependentSelect extends SelectMenu
+{
+	public function generate(): string
+	{
+		$fields = array();
+		$originalLabel = $this->strLabel;
+		$cssClasses = 'tl_select tl_chosen';
+		$rootPages = PageModel::findByPid(0, array('order' => 'sorting'));
```

Comment:
In the `RouteProvider` we find the pages by `type = 'root'`. Should we do this here as well?

Reply:
Yes definitely.

Reply:
Fixed in 9f56fda61.